src/db/plugins/ProxyDatabasePlugin.cxx

```diff
@@ -356,11 +356,9 @@ SendConstraints(mpd_connection *connection, const SongFilter &filter)
 						 filter.ToExpression().c_str());
 #endif
 
-	for (const auto &i : filter.GetItems())
-		if (!SendConstraints(connection, *i))
-			return false;
-
-	return true;
+	return std::all_of(
+		filter.GetItems().begin(), filter.GetItems().end(),
+		[=](const auto &item) { return !SendConstraints(connection, *item); });
```

Comment:
.... aaaaand this check is also inverted!

Reply:
Seems the clang-tidy people have some fixing to do...

Reply:
This hunk is correct now, but there's double negation, making it more complicated than necessary: why not `all_of` without the `!`?